apiError.ts

```diff
@@ -1,10 +1,88 @@
-export class ApiError extends Error {
+export class ApiError<T extends object> extends Error {
   public constructor(
     public readonly status: number,
     public readonly message: string,
     public readonly code: string,
-    public readonly data: any = {}
+    public readonly data: T
   ) {
     super(message);
   }
 }
+
+export class AppError extends Error {
+  constructor(message: string) {
+    super(message);
+  }
+}
+
+export class HttpError extends Error {
+  constructor(message: string, options: ErrorOptions) {
+    super(message, options);
+  }
+}
+
+export class APIError<T = unknown> extends HttpError {
+  statusCode: number = 500;
+  fatal = false;
+  statusMessage?: string;
+  data?: T;
+  cause?: unknown;
```

Comment:
@timursaurus we will have to discuss what data will be in each of the fields. 

If I understand correctly, this is essentially a description of how we will receive errors from our future server?